test/jdk/javax/crypto/CryptoPermissions/InconsistentEntries.java

```diff
@@ -23,57 +23,87 @@
+    public static void main(String[] args) throws Exception {
+        if (!POLICY_DIR.toFile().exists()) {
+            Files.createDirectory(POLICY_DIR);
         }
 
-        File testpolicy = new File(JAVA_HOME + FS + "conf" + FS + "security" +
-                FS + "policy" + FS + "testlimited" + FS + "default_local.policy");
-        if (testpolicy.length() == 0) {
-            throw new RuntimeException("policy: default_local.policy does not exist or is empty");
+        Path targetFile = POLICY_DIR.resolve(POLICY_FILE.getFileName());
+        Files.copy(POLICY_FILE, targetFile, StandardCopyOption.REPLACE_EXISTING);
+
+        try {
+            test();
+        } finally {
+            Files.delete(targetFile);
         }
+    }
 
-        Security.setProperty("crypto.policy", "testlimited");
+    private static void test() throws Exception {
+        ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
+                InconsistentEntriesTest.class.getName()
+        );
 
-        try {
-            int maxKeyLen = Cipher.getMaxAllowedKeyLength("AES");
-            throw new RuntimeException("Should fail due to inconsistent entries in policy file");
-        } catch (ExceptionInInitializerError e) {
-            e.printStackTrace();
-            System.out.println("Test completed successfully");
+        Process proc = pb.start();
+        OutputAnalyzer output = new OutputAnalyzer(proc);
+        output.shouldContain("Test completed successfully");
+        System.out.println("Test passed.");
+    }
+
+    class InconsistentEntriesTest {
+        public static void main(String[] args) throws Exception {
+            String JAVA_HOME = System.getProperty("java.home");
+            String FS = System.getProperty("file.separator");
+            Path testlimited = Path.of(JAVA_HOME + FS + "conf" + FS + "security" +
+                    FS + "policy" + FS + "testlimited");
+            if (!Files.exists(testlimited)) {
+                throw new RuntimeException(
+                        "custom policy subdirectory: testlimited does not exist");
+            }
+
+            File testpolicy = new File(JAVA_HOME + FS + "conf" + FS + "security" +
+                    FS + "policy" + FS + "testlimited" + FS + "default_local.policy");
+            if (testpolicy.length() == 0) {
+                throw new RuntimeException(
+                        "policy: default_local.policy does not exist or is empty");
+            }
+
+            Security.setProperty("crypto.policy", "testlimited");
+
+            try {
+                int maxKeyLen = Cipher.getMaxAllowedKeyLength("AES");
+                throw new RuntimeException(
+                        "Should fail due to inconsistent entries in policy file");
+            } catch (ExceptionInInitializerError e) {
+                e.printStackTrace();
+                System.out.println("Test completed successfully");
```

Comment:
This test can be rewritten and use @Test(expectedException = ExceptionInInitializerError.class) instead of creating the ProcessBuilder to capture the output.